### ISSUE TITLE
rebar3 eunit fails on OTP 21.1

### ISSUE THREAD

**Issue by uwiger** [martinsumner/leveled#223]
Some sample output:

```
===> Performing EUnit tests...
.........=CRASH REPORT==== 7-Dec-2018::09:00:29.958959 ===
  crasher:
    initial call: leveled_bookie:init/1
    pid: <0.959.0>
    registered_name: []
    exception exit: no_root_path
      in function  gen_server:init_it/6 (gen_server.erl, line 358)
    ancestors: [<0.660.0>]
    message_queue_len: 0
    messages: []
    links: []
    dictionary: []
    trap_exit: false
    status: running
    heap_size: 233
    stack_size: 27
    reductions: 233
  neighbours:
```
and a bit later:

```
.........................F=ERROR REPORT==== 7-Dec-2018::09:01:38.280915 ===
** Generic server <0.1234.0> terminating 
** Last message in was {'$gen_cast',{release_snapshot,<0.1239.0>}}
** When Server state == {state,
                            [{1,
                              [{3,"../test/journal/journal_files/nursery_3",
                                <0.1237.0>,
                                {4,stnd,{o,<<"B">>,"Key4",null}}},
                               {1,"../test/journal/journal_files/nursery_1",
                                <0.1236.0>,
                                {2,stnd,{o,<<"B">>,"Key2",null}}}]}],
                            1,4,<0.1237.0>,[],
                            [{<0.1239.0>,1}],
                            "../test/journal",
                            {cdb_options,300000,undefined,undefined,true,sync},
                            <0.1235.0>,false,undefined,false,native,true,
                            undefined}
** Reason for termination == 
** {terminated,[{io,format,
                    [<0.1230.0>,
                     "2018-12-07T09:01:38.280 I0004 ~w Remaining number of journal snapshots is ~w~n",
                     [<0.1234.0>,0]],
                    []},
                {leveled_inker,handle_cast,2,
                               [{file,"/home/uwiger/tmp/test/leveled/src/leveled_inker.erl"},
                                {line,698}]},
                {gen_server,try_dispatch,4,
                            [{file,"gen_server.erl"},{line,637}]},
                {gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,711}]},
                {proc_lib,init_p_do_apply,3,
                          [{file,"proc_lib.erl"},{line,249}]}]}
```

Switching to OTP 20.3, the same code (latest master: [` bfddb53`](https://github.com/martinsumner/leveled/tree/bfddb53e31d23083c39e08cb0f87f1f7ace2d25d))  passes eunit tests.

**Comment by martinsumner** [comment]
I've recreated this, and will take a look.  I've not tested 21 so far, so there's plenty of issues for me to work through.

**Comment by martinsumner** [comment]
@russelldb - the test failing is the `leveled_inker:handle_down_test/0` that was added in as part of the process monitoring changes you made.  The `erlang:process_info/1` call is now receiving process info not undefined.  Any ideas?

**Comment by russelldb** [comment]
Does it fail on all OTPs? Or just 21? I can't tell from the reports above

**Comment by uwiger** [comment]
At Aeternity, we're still running OTP 20. We may well move to 21 soon(-ish), but it is not a high priority at this time.

@russelldb The test passes on OTP 20.3.

**Comment by martinsumner** [comment]
Yes it is just 21.0 or above.  In the readme for 21.0 the monitor has been changed to a truly async message.  Maybe some timing issues now?

**Comment by benoitc** [comment]
the error still exist on 21.2 btw

**Comment by martinsumner** [comment]
I assume I must have fixed this at one stage.  I've double-checked that on `develop-3.0` all tests pass on OTP 21.3.  Support available through to OTP 24 on the `develop-3.1` branch.
